coeus-impl/src/main/java/co/kuali/coeus/workflow/UnitAdministratorRoleAttribute.java

```diff
@@ -0,0 +1,153 @@
+package co.kuali.coeus.workflow;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.jdom.Document;
+import org.jdom.Element;
+import org.kuali.coeus.common.framework.unit.Unit;
+import org.kuali.coeus.common.framework.unit.UnitService;
+import org.kuali.coeus.common.framework.unit.admin.UnitAdministrator;
+import org.kuali.coeus.propdev.impl.person.ProposalPerson;
+import org.kuali.coeus.propdev.impl.person.ProposalPersonUnit;
+import org.kuali.coeus.sys.framework.service.KcServiceLocator;
+import org.kuali.rice.core.api.util.xml.XmlHelper;
+import org.kuali.rice.kew.api.identity.Id;
+import org.kuali.rice.kew.api.identity.PrincipalId;
+import org.kuali.rice.kew.api.rule.RoleName;
+import org.kuali.rice.kew.engine.RouteContext;
+import org.kuali.rice.kew.routeheader.DocumentContent;
+import org.kuali.rice.kew.rule.GenericRoleAttribute;
+import org.kuali.rice.kew.rule.QualifiedRoleName;
+
+@SuppressWarnings("unchecked")
+public class UnitAdministratorRoleAttribute extends GenericRoleAttribute {
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 2187135270043141363L;
+
+	private static final Log LOG = LogFactory.getLog(UnitAdministratorRoleAttribute.class);
+	private static final String HOMEUNIT = "homeUnit";
+	//private static final String UNITS = "units";
+	
+	// Default Values 
+	private boolean getUnits = true;
+	private String roleName = "Department Head";
+	private String unitAdministratorCode = "3";
+	
+	UnitService unitService;
+	
+	public List<String> getQualifiedRoleNames(String roleName,
+			DocumentContent documentContent) {
+		
+		//LOG.info("getQualifiedRoleNames");
+		
+		// RoleName!Unit Administrator Code!TRUE OF PARENT
+		if (roleName != null) {
+			String[] info = roleName.split("!");
+			//LOG.info("roleName: " + roleName);
+			
+			if (info.length != 3) {
+				LOG.warn("Not enough parameters in the roleName.");
+			} else {
+				this.roleName = info[0];
+				this.unitAdministratorCode = info[1];
+				if (info[2].toUpperCase().equals("FALSE")) {
+					getUnits = false;
+				}
```

Comment:
this looks bad. I do not know the details of why they are doing it this way but it looks bad. I am just noting these things here so we know what to clean up later. We do not need to fix everything now.